### PR TITLE
GSA - Fix PFEH not being removed under some conditions

### PR DESCRIPTION
**Pull request by PabstMirror**
- GSA object deleted - PFEH now passes radioID to disconnect, so we can still remove the pfeh
- Radio removed by script - sys_radio,getRadioObject will return nil

also fix mast item being lost if no room in inventory

### CODE REVIEW

File: addons/sys_gsa/fnc_disconnectServer.sqf [hunk 1]
```diff
@@ -4,8 +4,9 @@
  * Disconnects the ground spike antenna and re-connects the default radio antenna. Server Event.
  *
  * Arguments:
- * 0: Ground Spike Antenna <OBJECT>
+ * 0: Ground Spike Antenna (could be null) <OBJECT>
  * 1: Player or unit <OBJECT>
+ * 2: RadioID <STRING><OPTIONAL>
```
Comment (jonpas) [inline addons/sys_gsa/fnc_disconnectServer.sqf]:
```suggestion
 * 2: RadioID <STRING> (default: "")
```

File: addons/sys_gsa/fnc_disconnectServer.sqf [hunk 2]
```diff
@@ -16,18 +17,28 @@
  * Public: No
  */
 
-params ["_gsa", "_unit"];
+params ["_gsa", "_unit", ["_radioId", ""]];
 
 private _success = false;
-private _radioId = _gsa getVariable [QGVAR(connectedRadio), ""];
+
+if (_radioId == "") then { _radioId = _gsa getVariable [QGVAR(connectedRadio), ""]; };
```
Comment (jonpas) [inline addons/sys_gsa/fnc_disconnectServer.sqf]:
```suggestion
if (_radioId == "") then {
    _radioId = _gsa getVariable [QGVAR(connectedRadio), ""];
};
```

**Review by jonpas — CHANGES_REQUESTED** [review]
Nice!

**Review by jonpas — APPROVED** [review]

File: addons/sys_gsa/fnc_disconnectServer.sqf [hunk 3]
```diff
@@ -16,18 +17,30 @@
...
 
 private _success = false;
-private _radioId = _gsa getVariable [QGVAR(connectedRadio), ""];
+
+if (_radioId == "") then {
+    _radioId = _gsa getVariable [QGVAR(connectedRadio), ""];
+};
 if (_radioId isEqualTo "") exitWith {
-    ERROR("Empty unique radio ID");
+    ERROR_3("Empty unique radio ID %1:%2:%3",_gsa,_unit,_radioId);
```
Comment (TheMagnetar) [inline addons/sys_gsa/fnc_disconnectServer.sqf]: _radioId is empty, no need to log it.

```suggestion
    ERROR_2("Empty unique radio ID %1:%2",_gsa,_unit);
```

**Review by TheMagnetar — CHANGES_REQUESTED** [review]

**Review by TheMagnetar — APPROVED** [review]

**Pull request merged**